src/shared/cloudformation/activation.ts

```diff
@@ -33,27 +33,33 @@ export const TEMPLATE_FILE_EXCLUDE_PATTERN = /.*[/\\]\.aws-sam([/\\].*|$)/
 export async function activate(extensionContext: vscode.ExtensionContext): Promise<void> {
     try {
         const registry = new CloudFormationTemplateRegistry()
-        globals.templateRegistry = registry
+        globals.templateRegistry.cfn = registry
         await registry.addExcludedPattern(TEMPLATE_FILE_EXCLUDE_PATTERN)
         await registry.addWatchPattern(TEMPLATE_FILE_GLOB_PATTERN)
         await registry.watchUntitledFiles()
     } catch (e) {
         vscode.window.showErrorMessage(
             localize(
                 'AWS.codelens.failToInitialize',
-                'Failed to activate template registry. {0}} will not appear on SAM template files.',
+                'Failed to activate cloudformation template registry. {0} will not appear on SAM template files.',
                 getIdeProperties().codelenses
             )
         )
         getLogger().error('Failed to activate template registry', e)
         // This prevents us from breaking for any reason later if it fails to load. Since
         // Noop watcher is always empty, we will get back empty arrays with no issues.
-        globals.templateRegistry = new NoopWatcher() as unknown as CloudFormationTemplateRegistry
+        globals.templateRegistry.cfn = new NoopWatcher() as unknown as CloudFormationTemplateRegistry
     }
     // If setting it up worked, add it to subscriptions so it is cleaned up at exit
     extensionContext.subscriptions.push(
-        globals.templateRegistry,
-        Commands.register('aws.cloudFormation.newTemplate', () => createStarterTemplateFile(false)),
-        Commands.register('aws.sam.newTemplate', () => createStarterTemplateFile(true))
+        globals.templateRegistry.cfn,
+        Commands.register('aws.cloudFormation.newTemplate', () => {
+            const contents = createCloudFormationTemplateYaml(false)
+            createStarterTemplateFile(contents)
+        }),
+        Commands.register('aws.sam.newTemplate', () => {
+            const contents = createCloudFormationTemplateYaml(true)
+            createStarterTemplateFile(contents)
```

Comment:
```suggestion
            await createStarterTemplateFile(contents)
```
or
```suggestion
            return createStarterTemplateFile(contents)
```
Commands should `await` or `return` all Promises they create, otherwise we can't accurately handle errors among other things. Need to enable linting for this. Or I could write an async hook to capture dangling promises but that's a lot of work.